From 89dec9d0a026d5336d9283b3389adaddfd86c2d4 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Fri, 30 Nov 2018 20:29:50 +0100
Subject: [PATCH 1/1] test/py: replace deprecated item.get_marker().

With pytest 3.6.0 get_marker() is deprecated. See
https://docs.pytest.org/en/latest/mark.html#updating-code

To avoid errors in Travis replace it by get_closest_marker().

This reverts commit f6206f8587fc "travis: Ensure we use pytest 2.8.7".

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 .travis.yml         | 2 +-
 test/py/conftest.py | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/.travis.yml b/.travis.yml
index a061f02399..6ce2ef7fba 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -47,7 +47,7 @@ install:
  - cat ~/.buildman
  - virtualenv /tmp/venv
  - . /tmp/venv/bin/activate
- - pip install pytest==2.8.7
+ - pip install pytest
  - pip install python-subunit
  - grub-mkimage -o ~/grub_x86.efi -O i386-efi normal  echo lsefimmap lsefi lsefisystab efinet tftp minicmd
  - grub-mkimage -o ~/grub_x64.efi -O x86_64-efi normal  echo lsefimmap lsefi lsefisystab efinet tftp minicmd
diff --git a/test/py/conftest.py b/test/py/conftest.py
index 5c658b8c48..44050ca206 100644
--- a/test/py/conftest.py
+++ b/test/py/conftest.py
@@ -431,7 +431,7 @@ def setup_boardspec(item):
         Nothing.
     """
 
-    mark = item.get_marker('boardspec')
+    mark = item.get_closest_marker('boardspec')
     if not mark:
         return
     required_boards = []
@@ -459,7 +459,7 @@ def setup_buildconfigspec(item):
         Nothing.
     """
 
-    mark = item.get_marker('buildconfigspec')
+    mark = item.get_closest_marker('buildconfigspec')
     if not mark:
         return
     for option in mark.args:
@@ -487,7 +487,7 @@ def setup_requiredtool(item):
         Nothing.
     """
 
-    mark = item.get_marker('requiredtool')
+    mark = item.get_closest_marker('requiredtool')
     if not mark:
         return
     for tool in mark.args:
-- 
2.19.2

